apps/passport-client/src/localstorage.ts

```diff
@@ -31,10 +39,18 @@ export async function loadPCDs(): Promise<PCDCollection> {
   }
 
   const serializedCollection = window.localStorage[COLLECTION_KEY];
-  return await PCDCollection.deserialize(
+  const collection = await PCDCollection.deserialize(
     await getPackages(),
     serializedCollection ?? "{}"
   );
+
+  if (!validateAndLogStateErrors(undefined, undefined, collection, true)) {
+    console.log(
```

Comment:
I don't like that this code isn't able to do anything about it other than log, given that the code above explicitly creates an empty PCDCollection, which is never valid.  I think I'd rather have this function return `undefined` rather than empty in this case, but that would require a broader refactor to adjust for broken assumptions.

Would it be better to throw an exception here to stop further progress?  Or do you think the other validation points are sufficient to ensure that bad data coming from here will never propagate farther, and will eventually be reported to the user at a different point?

Reply:
I was a bit nervous making the code throw at this location. I think I would be more comfortable starting by just logging and seeing if it happens, then making it throw at some later point in time with another code change.

> Or do you think the other validation points are sufficient to ensure that bad data coming from here will never propagate farther, and will eventually be reported to the user at a different point?

I feel pretty good about our other validation points catching the client before it gets too far in the case that it has an empty pcd collection